src/map/map.ts

```diff
@@ -1200,53 +1258,133 @@ export class PropertiesMap {
+              marker.style.left = `${x - 10}px`;
             }
 
-            this._restyle({
-                'x': this._xValues(1),
-                'y': this._yValues(1),
-                'z': this._zValues(1),
+          const indexes = this._indexer.from_environment(this._currentList[guid]);
+          this.onselect(indexes, guid);
+        }
+      }
+
+    private _addMarker(guid: string, index: number = 0, color?: string) {
+      let marker = document.getElementById(`chsp-selected-${guid}`);
+      if (!this._is3D()) {
+        if (marker === null) {
+          const activeButton = document.getElementById(`chsp-activate-${guid}`);
+          if (activeButton !== null) {
+            marker = document.createElement('div');
+            marker.classList.add('chsp-selected', 'chsp-inactive-map');
+            if (guid === this._active) {
+              marker.classList.toggle('chsp-inactive-map', false);
+              marker.classList.toggle('chsp-active', true);
+            }
+            marker.id = `chsp-selected-${guid}`;
+            marker.onclick = () => {this._setActive(guid); };
+
+            color = activeButton.style.backgroundColor;
+            marker.style.backgroundColor = color;
+            this._root.appendChild(marker);
+          }
+        } else {this._root.appendChild(marker); }
+      }
+      if (! this._guids.includes(guid)) {
+        this._guids.push(guid);
+      }
+      this._currentList[guid] = Math.max(0, index);
+    }
+    private _removeMarker(guid: string, removeGUID: boolean = true) {
+        const marker = document.getElementById(`chsp-selected-${guid}`);
+        if (marker !== null) {
+            const markerParent = marker.parentNode;
+            if (markerParent !== null) {
+              markerParent.removeChild(marker);
+            }
 
-                'marker.symbol': symbol,
-            } as Data, 1);
-        } else {
-            const xaxis = this._plot._fullLayout.xaxis;
-            const yaxis = this._plot._fullLayout.yaxis;
-
-            const computeX = (data: number) => xaxis.l2p(data) + xaxis._offset;
-            const computeY = (data: number) => yaxis.l2p(data) + yaxis._offset;
-
-            const x = computeX(this._xValues(0)[0][this._selected]);
-            const y = computeY(this._yValues(0)[0][this._selected]);
-
-            // hide the point if it is outside the plot, allow for up to 10px
-            // overflow (for points just on the border)
-            const minX = computeX(xaxis.range[0]) - 10;
-            const maxX = computeX(xaxis.range[1]) + 10;
-            const minY = computeY(yaxis.range[1]) - 10;
-            const maxY = computeY(yaxis.range[0]) + 10;
-            if (!isFinite(x) || !isFinite(y) || x < minX || x > maxX || y < minY || y > maxY) {
-                this._selectedMarker.style.display = 'none';
-            } else {
-                this._selectedMarker.style.display = 'block';
+            if (removeGUID) {
```

Comment:
In which case would we want to remove the marker but not the GUID?

Reply:
If you want to go to 3D, you need to remove the marker from the scene, but you want it cached in case you go back from 3D to 2D. There might be a better way to do this, I'll think about it.